### Чтение файлов в Pandas. Практическое задание

В этом практическом задании две обязательные задачи.

*Обязательные задачи* нужно сделать, чтобы проверить, что вы действительно усвоили материал урока. Если у вас всё получилось, вы можете переходить к следующему уроку.

Удачи!

Цели практического задания:

1. Научиться загружать данные из файлов форматов xlsx и csv.
2. Отработать обращение к элементам датафрейма по индексам и названиям строк и столбцов.

В конце ноутбука вы можете найти ответы для самостоятельной проверки.




---


**Обязательные задачи**

Вы будете работать с частью набора рекламных данных, показывающих, щёлкнул ли конкретный пользователь рекламу на сайте компании или нет.

Описание датасета:
- **Number**: Номер пользователя в системе.
- **Daily Time Spent on Site**: Среднее ежедневное время нахождения пользователя на сайте в минутах.
- **Daily Internet Usage**: Среднее ежедневное время нахождения пользователя в интернете в минутах.
- **Ad Topic Line**: Заголовок рекламного объявления.
- **Clicked on Ad**: Кликнул пользователь на рекламу или нет.

Даны файлы двух разных форматов с некоторыми данными для датасета выше:

- df4.xlsx.
- df5.csv.

Загрузите в отдельный датафрейм каждый из них, передав верные параметры там, где требуется:

**Задача 1**

1. Загрузите в df4 данные из xlsx-файла и выведите на экран первые пять строк. Не забудьте указать верный параметр header для отображения корректных заголовков.


In [1]:
# подключаемся к google-диску
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
# выбираем текущую директорию
import os
os.chdir('/content/drive/MyDrive/SkillBox/Pandas')

In [3]:
ls -l

total 1378
-rw------- 1 root root 578365 Jul 19 14:14 '12.1 Pandas.pdf'
-rw------- 1 root root  84562 Jul 20 14:04  12.1_SkillBox_Pandas.ipynb
-rw------- 1 root root  51339 Jul 20 14:32  12.1_SkillBox_Pandas_Практика.ipynb
drwx------ 2 root root   4096 Jul 20 12:19  12.2.Data/
-rw------- 1 root root  15720 Jul 21 22:24  12.2_Чтение_файлов_в_Pandas_Практика.ipynb
-rw------- 1 root root 214508 Jul 21 22:19  12.2_Чтение_файлов_с_помощью_Pandas.ipynb
-rw------- 1 root root 460387 Jul 20 08:23 '12.2 Чтение файлов с помощью Pandas.pdf'


In [4]:
import pandas as pd

In [30]:
# все файлы лежат в каталоге 12.2.Data/
# Допишите код
df4 = pd.read_excel('12.2.Data/df4.xlsx', header = 1)
df4

Number  Daily Time Spent On Site  DailyInternet Usage  \
0        5                     68.95               256.09   
1        6                     80.23               193.77   
2        7                     69.47               236.50   
3        8                     74.15               245.89   
4        9                       NaN               225.58   
5       10                     59.99               226.74   
6       11                     88.91               208.36   
7       12                       NaN               131.76   
8       13                     74.53                  NaN   
9       14                     69.88               183.82   
10      15                     47.64               122.02   
11      16                     83.07               230.87   
12      17                     69.57               113.12   
13      18                     79.52               214.23   
14      19                       NaN               143.56   
15      20                     63.45               140.64   
16      21                     55.39                  NaN   
17      22                     82.03               187.53   
18      23                     54.70               118.39   
19      24                     74.58               135.51   
20      25                     77.22               224.44   
21      26                     84.59               226.54   
22      27                       NaN               164.83   
23      28                     87.29               209.93   
24      29                     41.39                  NaN   
25      30                     78.74               204.79   
26      31                     48.53               134.14   
27      32                     51.95               129.23   
28      33                     70.20               119.20   
29      34                       NaN               209.82   

                                    Ad Topic Line  Clicked On Ad  
0              Cloned 5thgeneration orchestration              0  
1              Monitored national standardization              0  
2                Organic bottom-line service-desk              0  
3           Triple-buffered reciprocal time-frame              0  
4                   Robust logistical utilization              0  
5                 Sharable client-driven software              0  
6                      Enhanced dedicated support              0  
7                        Reactive local challenge              1  
8                  Configurable coherent function              0  
9              Mandatory homogeneous architecture              0  
10                 Centralized neutral neural-net              1  
11  Team-oriented grid-enabled Local Area Network              0  
12          Centralized content-based focus group              1  
13               Synergistic fresh-thinking array              0  
14                  Grass-roots coherent extranet              1  
15             Persistent demand-driven interface              1  
16             Customizable multi-tasking website              1  
17                     Intuitive dynamic attitude              0  
18   Grass-roots solution-oriented conglomeration              1  
19                     Advanced 24/7 productivity              1  
20          Object-based reciprocal knowledgebase              0  
21              Streamlined non-volatile analyzer              0  
22          Mandatory disintermediate utilization              1  
23             Future-proofed methodical protocol              0  
24                  Exclusive neutral parallelism              1  
25                Public-key foreground groupware              0  
26             Ameliorated client-driven forecast              1  
27                 Monitored systematic hierarchy              1  
28        Open-architected impactful productivity              1  
29        Business-focused value-added definition              0

In [31]:
# Ваш код здесь
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Number                    30 non-null     int64  
 1   Daily Time Spent On Site  25 non-null     float64
 2   DailyInternet Usage       27 non-null     float64
 3   Ad Topic Line             30 non-null     object 
 4   Clicked On Ad             30 non-null     int64  
dtypes: float64(2), int64(2), object(1)
memory usage: 1.3+ KB


2. Найдите в датафрейме df4 значение 193.77 и выведите его на экран, используя выбор элементов по индексу.

In [32]:
# Ваш код c iloc здесь
df4.iloc[1,2]

193.77

**Задача 2**

1. Загрузите в df5 данные из csv-файла и выведите на экран последние десять строк. Не забудьте указать верный разделитель.


In [18]:
# Допишите код
df5 = pd.read_csv('12.2.Data/df5.csv', sep = '|')
df5

Number  Daily Time Spent on Site  Daily Internet Usage  \
0       30                     78.74                204.79   
1       31                     48.53                134.14   
2       32                     51.95                129.23   
3       33                     70.20                119.20   
4       34                       NaN                209.82   
..     ...                       ...                   ...   
80     110                     80.49                229.12   
81     111                     72.23                241.03   
82     112                     42.39                150.99   
83     113                       NaN                135.18   
84     114                     74.02                210.54   

                                   Ad Topic Line  Clicked on Ad  
0                Public-key foreground groupware              0  
1             Ameliorated client-driven forecast              1  
2                 Monitored systematic hierarchy              1  
3        Open-architected impactful productivity              1  
4        Business-focused value-added definition              0  
..                                           ...            ...  
80                   Balanced heuristic approach              0  
81                 Focused 24hour implementation              0  
82           De-engineered mobile infrastructure              1  
83  Customer-focused upward-trending contingency              1  
84              Operative system-worthy protocol              0  

[85 rows x 5 columns]

In [19]:
# Ваш код здесь
df5.tail(10)

Number  Daily Time Spent on Site  Daily Internet Usage  \
75     105                     41.49                169.18   
76     106                     63.60                235.28   
77     107                     89.91                194.23   
78     108                     68.18                218.17   
79     109                       NaN                202.16   
80     110                     80.49                229.12   
81     111                     72.23                241.03   
82     112                     42.39                150.99   
83     113                       NaN                135.18   
84     114                     74.02                210.54   

                                   Ad Topic Line  Clicked on Ad  
75               Persevering reciprocal firmware              1  
76           Centralized logistical secured line              0  
77          Innovative background conglomeration              0  
78                  Switchable 3rdgeneration hub              0  
79        Polarized 6thgeneration info-mediaries              0  
80                   Balanced heuristic approach              0  
81                 Focused 24hour implementation              0  
82           De-engineered mobile infrastructure              1  
83  Customer-focused upward-trending contingency              1  
84              Operative system-worthy protocol              0

2. Найдите в датафрейме df5 значение `Polarized 6thgeneration info-mediaries` и выведите его на экран, используя выбор элементов по названию.


In [24]:
# Ваш код c loc здесь
df5.loc[79,'Ad Topic Line']

'Polarized 6thgeneration info-mediaries'

In [28]:
df5[df5.loc[:,'Ad Topic Line'] == 'Polarized 6thgeneration info-mediaries']

Number  Daily Time Spent on Site  Daily Internet Usage  \
79     109                       NaN                202.16   

                             Ad Topic Line  Clicked on Ad  
79  Polarized 6thgeneration info-mediaries              0



|

|

|

|

|

|

|

|

|

|

|

|

|

|

|

|

|

|

**Ответы для самостоятельной проверки**

**Задача 1**

In [ ]:
# 1.

df4 = pd.read_excel('df4.xlsx', header=1)
df4.head()

Number  Daily Time Spent on Site  Daily Internet Usage  \
0       5                     68.95                256.09   
1       6                     80.23                193.77   
2       7                     69.47                236.50   
3       8                     74.15                245.89   
4       9                       NaN                225.58   

                           Ad Topic Line  Clicked on Ad  
0     Cloned 5thgeneration orchestration              0  
1     Monitored national standardization              0  
2       Organic bottom-line service-desk              0  
3  Triple-buffered reciprocal time-frame              0  
4          Robust logistical utilization              0

In [ ]:
# 2.

df4.iloc[1, 2]

193.77

**Задача 2**

In [ ]:
# 1.

df5 = pd.read_csv('df5.csv', sep='|')
df5.tail(10)

Number  Daily Time Spent on Site  Daily Internet Usage  \
75     105                     41.49                169.18   
76     106                     63.60                235.28   
77     107                     89.91                194.23   
78     108                     68.18                218.17   
79     109                       NaN                202.16   
80     110                     80.49                229.12   
81     111                     72.23                241.03   
82     112                     42.39                150.99   
83     113                       NaN                135.18   
84     114                     74.02                210.54   

                                   Ad Topic Line  Clicked on Ad  
75               Persevering reciprocal firmware              1  
76           Centralized logistical secured line              0  
77          Innovative background conglomeration              0  
78                  Switchable 3rdgeneration hub              0  
79        Polarized 6thgeneration info-mediaries              0  
80                   Balanced heuristic approach              0  
81                 Focused 24hour implementation              0  
82           De-engineered mobile infrastructure              1  
83  Customer-focused upward-trending contingency              1  
84              Operative system-worthy protocol              0

In [ ]:
# 2.

df5.loc[79, 'Ad Topic Line']

'Polarized 6thgeneration info-mediaries'